src/v2/network/client.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+export const client = {
+	createContactGroup: (title: string, members: Array<string>): Promise<any> =>
+		fetch(`/service/soap/CreateContactRequest`, {
+			method: 'POST',
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			body: JSON.stringify({
+				Body: {
+					[`CreateContactRequest`]: {
+						_jsns: 'urn:zimbraMail',
+						cn: {
+							a: [
+								{ n: 'fullName', _content: title },
+								{ n: 'nickname', _content: title },
+								{ n: 'type', _content: 'group' },
+								{ n: 'fileAs', _content: `8:${title}` }
+							],
+							m: members.map((member) => ({ type: 'I', value: member }))
+						}
+					}
+				},
+				Header: {
+					context: {
+						_jsns: 'urn:zimbra'
+					}
+				}
+			})
+		}).then((response) => {
+			if (response.ok) {
+				return response.json();
+			}
+			throw new Error('Something went wrong');
```

Comment:
It could be worth it trying to throw the Fault message as error message instead of the generic message